### ISSUE TITLE
Upgrade to Hibernate 6.5.3

### ISSUE THREAD

**Issue by christophstrobl** [spring-projects/spring-data-jpa#3473]
The hibernate 6.5.1 upgrade comes with a different Bytecode Provider [setup](https://github.com/hibernate/hibernate-orm/commit/aff789cd903e830bd5e01fd0307bcb66d6d0657c#diff-4f7a1f86638dba1180629a9abe02eb3074e64d75fd181f9462e2b13ebfbd3048), that breaks in combination with `spring-instrument` leading to `LinkageError` errors as outlined further down below.

```
java.lang.LinkageError: loader 'app' attempted duplicate class definition for org.hibernate.annotations.common.reflection.java.generics.IdentityTypeEnvironment. (org.hibernate.annotations.common.reflection.java.generics.IdentityTypeEnvironment is in unnamed module of loader 'app')
	at org.hibernate.annotations.common.reflection.java.JavaReflectionManager.toXClass(JavaReflectionManager.java:85)
	at org.hibernate.boot.model.source.internal.annotations.AnnotationMetadataSourceProcessorImpl.toXClass(AnnotationMetadataSourceProcessorImpl.java:175)
	at org.hibernate.boot.model.source.internal.annotations.AnnotationMetadataSourceProcessorImpl.<init>(AnnotationMetadataSourceProcessorImpl.java:97)
	at org.hibernate.boot.model.process.spi.MetadataBuildingProcess$1.<init>(MetadataBuildingProcess.java:206)
	at org.hibernate.boot.model.process.spi.MetadataBuildingProcess.complete(MetadataBuildingProcess.java:200)
	at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.metadata(EntityManagerFactoryBuilderImpl.java:1431)
	at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.build(EntityManagerFactoryBuilderImpl.java:1502)
	at org.springframework.orm.jpa.vendor.SpringHibernateJpaPersistenceProvider.createContainerEntityManagerFactory(SpringHibernateJpaPersistenceProvider.java:75)
	at org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean.createNativeEntityManagerFactory(LocalContainerEntityManagerFactoryBean.java:390)
	at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.buildNativeEntityManagerFactory(AbstractEntityManagerFactoryBean.java:409)
	at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.afterPropertiesSet(AbstractEntityManagerFactoryBean.java:396)
	at org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean.afterPropertiesSet(LocalContainerEntityManagerFactoryBean.java:366)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1833)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1782)
```

**Comment by mp911de** [comment]
Related ticket in Hibernate: https://hibernate.atlassian.net/browse/HHH-18108

**Comment by christophstrobl** [comment]
The issue has been solved for hibernate 6.5.3

**Comment by mp911de** [comment]
Superseded by #3574
